discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,191 @@
+	Zone string `yaml:"zone"`
+	// AccessKey used to authenticate on Scaleway APIs.
+	AccessKey string `yaml:"access_key"`
+	// SecretKey used to authenticate on Scaleway APIs.
+	SecretKey config.Secret `yaml:"secret_key"`
+	// FilterName to filter on during the ListServers.
+	FilterName string `yaml:"name"`
+	// FilterTags to filter on during the ListServers.
+	FilterTags []string `yaml:"tags"`
+
+	HTTPClientConfig config.HTTPClientConfig `yaml:",inline"`
+
+	RefreshInterval model.Duration `yaml:"refresh_interval"`
+	Port            int            `yaml:"port"`
+	// Role can be either instance or baremetal
+	Role role `yaml:"role"`
+}
+
+func (c SDConfig) Name() string {
+	return "scaleway"
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultSDConfig
+	type plain SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+
+	if c.Role == "" {
+		return errors.New("role missing (one of: instance, baremetal)")
+	}
+
+	return c.HTTPClientConfig.Validate()
+}
+
+func (c SDConfig) NewDiscoverer(options discovery.DiscovererOptions) (discovery.Discoverer, error) {
+	return NewDiscovery(&c, options.Logger)
+}
+
+// SetDirectory joins any relative file paths with dir.
+func (c *SDConfig) SetDirectory(dir string) {
+	c.HTTPClientConfig.SetDirectory(dir)
+}
+
+func init() {
+	discovery.RegisterConfig(&SDConfig{})
+}
+
+// Discovery periodically performs Scaleway requests. It implements
+// the Discoverer interface.
+type Discovery struct {
+}
+
+func NewDiscovery(conf *SDConfig, logger log.Logger) (*refresh.Discovery, error) {
+	r, err := newRefresher(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return refresh.NewDiscovery(
+		logger,
+		"scaleway",
+		time.Duration(conf.RefreshInterval),
+		r.refresh,
+	), nil
+}
+
+type refresher interface {
+	refresh(context.Context) ([]*targetgroup.Group, error)
+}
+
+func newRefresher(conf *SDConfig) (refresher, error) {
+	switch conf.Role {
+	case roleBaremetal:
+		return newBaremetalDiscovery(conf)
+	case roleInstance:
+		return newInstanceDiscovery(conf)
+	}
+	return nil, errors.New("unknown Scaleway discovery role")
+}
+
+func LoadProfile(sdConfig *SDConfig) (*scw.Profile, error) {
+	// By default we set default zone and region to fr-par
+	defaultZoneProfile := &scw.Profile{
+		DefaultRegion: scw.StringPtr(scw.RegionFrPar.String()),
+		DefaultZone:   scw.StringPtr(scw.ZoneFrPar1.String()),
+	}
+
+	// Profile coming from Prometheus Configuration file
+	prometheusConfigProfile := &scw.Profile{}
+
+	if sdConfig.AccessKey != "" {
+		prometheusConfigProfile.AccessKey = scw.StringPtr(sdConfig.AccessKey)
+	}
+	if sdConfig.SecretKey != "" {
+		prometheusConfigProfile.SecretKey = scw.StringPtr(string(sdConfig.SecretKey))
+	}
+	if sdConfig.Project != "" {
+		prometheusConfigProfile.DefaultProjectID = scw.StringPtr(sdConfig.Project)
+	}
+	if sdConfig.Zone != "" {
```

Comment:
```suggestion
	if sdConfig.Zone != "" {
```

If we add the Zone to DefaultSDConfig, this will never be empty.